### PR TITLE
[KBFS-1515] Refactor tests which use a fake block server

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: libkbfs/bserver_remote_test.go [hunk 1]
```diff
@@ -5,277 +5,115 @@
...
-
-	tlfID, err := tlf.ParseID(arg.Folder)
-	if err != nil {
-		return err
-	}
-
-	serverHalf, err := kbfscrypto.ParseBlockCryptKeyServerHalf(arg.BlockKey)
-	if err != nil {
-		return err
+	refs[[8]byte{}] = keybase1.BlockReference{
```
Comment (strib) [inline libkbfs/bserver_remote_test.go]: Why not use `kbfsblock.ZeroRefNonce`?

**Comment by akalin-keybase** [comment]
I'm not entirely happy about moving `BSR` into `kbfsblock` either, but I don't want to lose the bserver integration test with the client in the future world where bserver doesn't depend on libkbfs.

The separation between the implementation and the interface isn't weird in Go -- I think the ability to do that is by design. Another possibility is to simply move the interface into `kbfsblock`, too.

I also thought of `kbfsblock` as a data format package, but if we define it as "the set of things the bserver needs from the kbfs repo", then it makes more sense as to why `BlockServerRemote` is in there, and not `BlockServerMemory`.

I think this PR is still worth merging, though.

Thoughts on how to solve the BSR problem?

**Comment by strib** [comment]
> if we define it as "the set of things the bserver needs from the kbfs repo"

This definition doesn't sit right with me since it's an open source repo, and consumers of this repo shouldn't have to care about what the server implementation is like.  The splits should make sense based on just what's going on in the repo itself, whenever possible.

> I think this PR is still worth merging, though.

Agreed.

> Thoughts on how to solve the BSR problem?

I was thinking we'd change the bserver integration test to use either something that implements the {{BlockServer}} interface by calling the bserver directly.  I'm not sure there's much value in testing the RPC flow, since we now have kbfsdocker (we didn't used to ).

If you think it's really important we keep those tests going through real RPC, I guess I'd argue for moving it to a different package, `kbfs/bserver` or something.  Either with that or with `kbfsblock`, we'd have to move the `currentInfoGetter` definition out of `libkbfs`.

**Comment by akalin-keybase** [comment]
Ah, right, kbfsdocker does remove the need to test the RPC flow. I'll work towards removing it from the bserver integration test, then!

File: libkbfs/bserver_remote_test.go (hunk 1 above)
Comment (akalin-keybase) [inline libkbfs/bserver_remote_test.go]: wrong type -- but `keybase1.BlockReference{}` works so did that

**Pull request merged**